### PR TITLE
llama: Add support for RWKV v7 architecture(v2)

### PR DESCRIPTION
**Pull request by MollySophia**
@BlinkDL 's explanation of RWKV v7:
[RWKV-7 as a meta-in-context learner](https://x.com/BlinkDL_AI/status/1861753903886561649)
Also there are plenty of tests on trained models posted on his x account.

Current available RWKV v7 model repos in HF format:
### Base models:
https://huggingface.co/fla-hub/rwkv7-191M-world
https://huggingface.co/fla-hub/rwkv7-0.4B-world
https://huggingface.co/fla-hub/rwkv7-1.5B-world
https://huggingface.co/fla-hub/rwkv7-2.9B-world
https://huggingface.co/fla-hub/rwkv7-0.1B-g1 (Haven't add the option to enable it's <think> capability yet.)
### Distilled models:
https://huggingface.co/RWKV-Red-Team/ARWKV-R1-1B5
https://huggingface.co/RWKV-Red-Team/ARWKV-R1-7B
https://huggingface.co/RWKV-Red-Team/ARWKV_7B_R1_16K

This PR contains:
- `GGML_OP_L2_NORM` that applies [pytorch-style](https://pytorch.org/docs/stable/generated/torch.nn.functional.normalize.html) l2 normalization, along the rows. Tested with CPU, CUDA, SYCL, Vulkan, Metal backends.
- `GGML_OP_RWKV_WKV7` which is the core of the RWKV v7 architecture. Implemented the naive recurrent wkv7 kernel in CPU, CUDA, SYCL, Vulkan, Metal.
- Support inference of RWKV7 and ARWKV7 models.
- Simple Metal kernel for the old WKV6.
- Skip unused tokens in last layer ffn computation for rwkv models.
- Fix inference with RWKV6Qwen2.

TODO:
- [x] llama-parallel seems broken with all rwkv models. Will check what's wrong and try to fix them tomorrow. (Inference is fixed. But the output seems mixed between these parallel sequences. Haven't figured out what's wrong yet)
- [x] Why is Musa build failing? (Seems that there's some bugs in their vectorization code. Getting rid of a `#pragma unroll` in wkv.cu fix the build.

### CODE REVIEW

**Review by ggerganov — APPROVED** [review]

**Review by Rbiessy — APPROVED** [review]
No concern with the SYCL changes, thanks
